bigchaindb/common/schema/__init__.py

```diff
@@ -7,18 +7,46 @@
 from bigchaindb.common.exceptions import SchemaValidationError
 
 
-TX_SCHEMA_PATH = os.path.join(os.path.dirname(__file__), 'transaction.yaml')
-with open(TX_SCHEMA_PATH) as handle:
-    TX_SCHEMA_YAML = handle.read()
-TX_SCHEMA = yaml.safe_load(TX_SCHEMA_YAML)
-
-
-def validate_transaction_schema(tx_body):
-    """ Validate a transaction dict against a schema """
+def drop_schema_descriptions(node):
+    """ Drop descriptions from schema, since they clutter log output """
+    if 'description' in node:
+        del node['description']
+    for n in node.get('properties', {}).values():
+        drop_schema_descriptions(n)
+    for n in node.get('definitions', {}).values():
+        drop_schema_descriptions(n)
+    for n in node.get('anyOf', []):
+        drop_schema_descriptions(n)
+
+
+def _load_schema(name):
+    """ Load a schema from disk """
+    path = os.path.join(os.path.dirname(__file__), name + '.yaml')
+    with open(path) as handle:
+        schema = yaml.safe_load(handle)
+    drop_schema_descriptions(schema)
+    return path, schema
+
+
+def _validate_schema(schema, body):
+    """ Validate data against a schema """
     try:
-        jsonschema.validate(tx_body, TX_SCHEMA)
+        jsonschema.validate(body, schema)
     except jsonschema.ValidationError as exc:
         raise SchemaValidationError(str(exc))
```

Comment:
Oh yeah, we can also reraise the original to keep the stacktrace: `raise SchemaValidationError(str(exc)) from exc`